### PR TITLE
Add coupled Allen-Cahn kernel (and coupled Reaction)

### LINKED ISSUE #6194: Add CoupledAllenCahn kernel
A `CoupledAllenCahn` kernel will allow the implementation of generalized Gizburg-Landau equations in MOOSE. It seems to me that a generalized Gizburg-Landau equation is basically an Allen-Cahn equation with terms on the right side for the functional derivatives w.r.t. to all order parameters and a summation over the Onsager coefficients (whereas Allen-Cahn typically assomes the Onsager matrix to be diagonal).

Ginzburg-Landau

The right-hand side terms include Allen-Cahn terms for coupled variables for i!=j

Ping @laagesen, @tonkmr

### PR DESCRIPTION
**Pull request by dschwen**
This adds the `CoupledAllenCahn` kernel that will allow constructing systems with non-diagonal Onsager matrices. The test returns the same result as the non-coupled AC. I added an additional variable and coupled them using the Reaction kernel (R_v=v) and the new `CoupledReaction` kernel (R_eta=-v). The coupled reaction kernel needs the negative sign for this to form a valid substitution. 

I also continued with my crusade of renaming tests.

Refs #6194

### CODE REVIEW

File: modules/phase_field/include/kernels/CoupledAllenCahn.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+//Forward Declarations
+class CoupledAllenCahn;
+
+template<>
+InputParameters validParams<CoupledAllenCahn>();
+
+/**
+ * CoupledAllenCahn uses the Free Energy function and derivatives
+ * provided by a DerivativeParsedMaterial to computer the
```
Comment (laagesen) [inline modules/phase_field/include/kernels/CoupledAllenCahn.h]: typo: computer -> compute

**Comment by dschwen** [comment]
The next PR will add a `CoupledACInterface` class.

**Comment by laagesen** [comment]
just the one typo, otherwise looks good.

File: modules/phase_field/tests/phase_field_kernels/CoupledAllenCahn.i [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  nz = 0
+  xmin = 0
+  xmax = 50
+  ymin = 0
+  ymax = 50
+  zmin = 0
+  zmax = 50
+  elem_type = QUAD4
+
+  uniform_refine = 1
```
Comment (permcody) [inline modules/phase_field/tests/phase_field_kernels/CoupledAllenCahn.i]: What's the runtime of these tests? 1600 elements is fairly large

Comment (dschwen) [inline modules/phase_field/tests/phase_field_kernels/CoupledAllenCahn.i]: Less than second. And it reuses the gold file of an existing test.

Comment (permcody) [inline modules/phase_field/tests/phase_field_kernels/CoupledAllenCahn.i]: So fast that you left at the article "a" in that description!

Comment (dschwen) [inline modules/phase_field/tests/phase_field_kernels/CoupledAllenCahn.i]: Does not compute

Comment (permcody) [inline modules/phase_field/tests/phase_field_kernels/CoupledAllenCahn.i]: ... and I typed that so fast that I failed to follow normal grammar rules in my response to your incorrect grammar usage.

**Pull request merged**